WAIT_CLASS panic

Script:
```
LET sports = DOCUMENT('https://www.sports.ru/football/match/2017-01-01', true)
WAIT_CLASS(sports, '.panel active-panel js-active')
FOR elem IN ELEMENTS(sports, '.panel active-panel js-active') 
    RETURN elem
```
Output:
```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/MontFerret/ferret/pkg/stdlib/html.WaitClass(0x162b540, 0xc00036aab0, 0xc000108840, 0x2, 0x2, 0xc0000ab510, 0x0, 0x0, 0xc00032ebb0)
	/Users/user/go/src/github.com/MontFerret/ferret/pkg/stdlib/html/wait_class.go:47 +0x79e
github.com/MontFerret/ferret/pkg/runtime/expressions.(*FunctionCallExpression).Exec(0xc0002fdb80, 0x162b540, 0xc00036aab0, 0xc00036a900, 0x16301c0, 0x1a9c140, 0x0, 0x0)
	/Users/user/go/src/github.com/MontFerret/ferret/pkg/runtime/expressions/func_call.go:63 +0x27a
github.com/MontFerret/ferret/pkg/runtime/expressions.(*BlockExpression).Exec(0xc00036a6f0, 0x162b540, 0xc00036aab0, 0xc00036a900, 0xc00036aa20, 0x20, 0x14fc620, 0x1056001)
	/Users/user/go/src/github.com/MontFerret/ferret/pkg/runtime/expressions/block.go:49 +0x83
github.com/MontFerret/ferret/pkg/runtime.(*Program).Run(0xc0003415c0, 0x162b480, 0xc0002fdc00, 0xc0002c5d38, 0x6, 0x6, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/user/go/src/github.com/MontFerret/ferret/pkg/runtime/program.go:40 +0x1c1
github.com/MontFerret/ferret/cli.Exec(0xc0001760e0, 0xd2, 0x7ffeefbffb9d, 0x15, 0xc00019c7e0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/user/go/src/github.com/MontFerret/ferret/cli/exec.go:60 +0x44f
github.com/MontFerret/ferret/cli.ExecFile(0x7ffeefbffbb3, 0xd, 0x7ffeefbffb9d, 0x15, 0xc00019c7e0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/user/go/src/github.com/MontFerret/ferret/cli/exec.go:24 +0x113
main.main()
	/Users/user/go/src/github.com/MontFerret/ferret/main.go:183 +0x29e
```

--- a/e2e/pages/dynamic/components/pages/events/clickable.js
+++ b/e2e/pages/dynamic/components/pages/events/clickable.js
@@ -1,0 +1,54 @@
+const e = React.createElement;
+
+export default class ClickableComponent extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            clicked: false
+        };
+    }
+
+    handleClick() {
+        let timeout = 500;
+
+        if (this.props.randomTimeout) {
+            timeout = Math.ceil(Math.random() * 1000 * 10);
+        }
+
+        setTimeout(() => {
+            this.setState({
+                clicked: !this.state.clicked
+            })
+        }, timeout)
+    }
+
+    render() {
+        const btnId = `${this.props.id}-btn`;
+        const contentId = `${this.props.id}-content`;
+        const classNames = ["alert"];
+
+        if (this.state.clicked) {
+            classNames.push("alert-success");
+        }
+
+        return e("div", {className: "card"}, [
+            e("div", { className: "card-header"}, [
+                e("button", {
+                    id: btnId,
+                    className: "btn btn-primary",
+                    onClick: this.handleClick.bind(this)
+                }, [
+                    "Toggle class"
+                ])
+            ]),
+            e("div", { className: "card-body"}, [
+                e("div", { id: contentId, className: classNames.join(" ")}, [
+                    e("p", null, [
+                        "Lorem ipsum dolor sit amet."
+                    ])
+                ])
+            ])
+        ]);
+    }
+}

--- a/e2e/pages/dynamic/components/pages/events/hoverable.js
+++ b/e2e/pages/dynamic/components/pages/events/hoverable.js
@@ -22,31 +22,26 @@ export default class HoverableComponent extends React.PureComponent {
     }
 
     render() {
-        const children = [];
-        children.push(
-            e("p", null, [
-                e("a", {
+        let content;
+
+        if (this.state.hovered) {
+            content =  e("p", { id: "hoverable-content"}, [
+                "Lorem ipsum dolor sit amet."
+            ]);
+        }
+
+        return e("div", { className: "card"}, [
+            e("div", {className: "card-header"}, [
+                e("button", {
                     id: "hoverable-btn",
                     className: "btn btn-primary",
-                    href: "#",
                     onMouseEnter: this.handleMouseEnter.bind(this),
                     onMouseLeave: this.handleMouseLeave.bind(this)
                 }, [
-                    "Hoverable link"
-                ]),
-            ])
-        );
-
-        if (this.state.hovered) {
-            children.push(
-                e("div", null, [
-                    e("div", { id: "hoverable-content", className: "card card-body"}, [
-                        "Lorem ipsum dolor sit amet."
-                    ])
+                    "Show content"
                 ])
-            )
-        }
-
-        return e("div", null, children);
+            ]),
+            e("div", {className: "card-body"}, content)
+        ]);
     }
 }

--- a/e2e/pages/dynamic/components/pages/events/index.js
+++ b/e2e/pages/dynamic/components/pages/events/index.js
@@ -1,12 +1,19 @@
 import Hoverable from "./hoverable.js";
+import Clickable from "./clickable.js";
 
 const e = React.createElement;
 
 export default class EventsPage extends React.Component {
     render() {
         return e("div", { className: "row", id: "page-events" }, [
-            e("div", { className: "col-lg-12"}, [
-                e(Hoverable)
+            e("div", { className: "col-lg-4"}, [
+                e(Hoverable),
+            ]),
+            e("div", { className: "col-lg-4"}, [
+                e(Clickable, { id: "wait-class" })
+            ]),
+            e("div", { className: "col-lg-4"}, [
+                e(Clickable, { id: "wait-class-random", randomTimeout: true })
             ])
         ])
     }

--- a/e2e/tests/doc_wait_class_d.fql
+++ b/e2e/tests/doc_wait_class_d.fql
@@ -1,0 +1,14 @@
+LET url = @dynamic + "?redirect=/events"
+LET doc = DOCUMENT(url, true)
+
+WAIT_ELEMENT(doc, "#page-events")
+
+// with fixed timeout
+CLICK(doc, "#wait-class-btn")
+WAIT_CLASS(doc, "#wait-class-content", "alert-success")
+
+// with random timeout
+CLICK(doc, "#wait-class-random-btn")
+WAIT_CLASS(doc, "#wait-class-random-content", "alert-success", 10000)
+
+RETURN ""

--- a/e2e/tests/el_wait_class_d.fql
+++ b/e2e/tests/el_wait_class_d.fql
@@ -1,0 +1,20 @@
+LET url = @dynamic + "?redirect=/events"
+LET doc = DOCUMENT(url, true)
+
+WAIT_ELEMENT(doc, "#page-events")
+
+// with fixed timeout
+LET b1 = ELEMENT(doc, "#wait-class-btn")
+LET c1 = ELEMENT(doc, "#wait-class-content")
+
+CLICK(b1)
+WAIT_CLASS(c1, "alert-success")
+
+// with random timeout
+LET b2 = ELEMENT(doc, "#wait-class-random-btn")
+LET c2 = ELEMENT(doc, "#wait-class-random-content")
+
+CLICK(b2)
+WAIT_CLASS(c2, "alert-success", 10000)
+
+RETURN ""

--- a/pkg/stdlib/html/wait_class.go
+++ b/pkg/stdlib/html/wait_class.go
@@ -43,6 +43,13 @@ func WaitClass(_ context.Context, args ...core.Value) (core.Value, error) {
 	// lets figure out what is passed as 1st argument
 	switch args[0].(type) {
 	case *dynamic.HTMLDocument:
+		// revalidate args with more accurate amount
+		err := core.ValidateArgs(args, 3, 4)
+
+		if err != nil {
+			return values.None, err
+		}
+
 		// class
 		err = core.ValidateType(args[2], core.StringType)
 
@@ -86,7 +93,7 @@ func WaitClass(_ context.Context, args ...core.Value) (core.Value, error) {
 				return values.None, err
 			}
 
-			timeout = args[3].(values.Int)
+			timeout = args[2].(values.Int)
 		}
 
 		return values.None, el.WaitForClass(class, timeout)

